pkg/fanal/artifact/image/image.go

```diff
@@ -240,27 +238,27 @@ func (a Artifact) inspect(ctx context.Context, missingImage string, layerKeys, b
 
 			layerInfo, err := a.inspectLayer(ctx, layer, disabledAnalyers)
 			if err != nil {
-				errCh <- xerrors.Errorf("failed to analyze layer (%s): %w", layer.DiffID, err)
-				return
+				return xerrors.Errorf("failed to analyze layer (%s): %w", layer.DiffID, err)
 			}
 			if err = a.cache.PutBlob(layerKey, layerInfo); err != nil {
-				errCh <- xerrors.Errorf("failed to store layer: %s in cache: %w", layerKey, err)
-				return
+				return xerrors.Errorf("failed to store layer: %s in cache: %w", layerKey, err)
 			}
 			if lo.IsNotEmpty(layerInfo.OS) {
 				osFound = layerInfo.OS
 			}
-		}(ctx, k)
+			return nil
+		})
+
+		if ctx.Err() != nil {
+			break
+		}
```

Comment:
I think it would be worse to remove it - there's no correctness problem with removing the check, but inspecting a layer can be quite expensive, so it seems like we should stop doing it if we already know that we're going to get an error result when we call `group.Wait()`